sources/packages/api/src/services/application/application.service.ts

```diff
@@ -324,16 +324,21 @@ export class ApplicationService extends RecordDataModelService<Application> {
     return application;
   }
 
+  /**
+   * get student application details with the application Id.
+   * @param applicationId student application id .
+   * @returns student Application Details.
+   */
   async getApplicationById(applicationId: number): Promise<Application> {
-    const application = this.repo
-      .createQueryBuilder("application")
-      .select(["application", "programYear.programYear"])
-      .innerJoin("application.programYear", "programYear")
-      .where("programYear.active = true")
-      .andWhere("application.id = :applicationId", {
-        applicationId,
-      });
-    return application.getOne();
+    return this.repo.findOne(applicationId, {
```

Comment:
Can we be more specific in the select clause here? It seems to me that we are retrieving way more than we need.